strings/base_includes.h

```diff
@@ -28,6 +28,10 @@
 #include <format>
 #endif
 
+#ifdef __cpp_lib_ranges
```

Comment:
Pre-existing: I suggest "pre-loading" the feature-test macros with:
```c++
#ifdef __has_include
#if __has_include(<version>)
#include <version>
#endif
#endif
```
before testing them like this. (MSVC's STL defines all feature-test macros when any STL header is included, but that behavior is not required by the C++ Standard and hence not portable.)

Reply:
Good idea, will add